images/ipfailover/keepalived/lib/failover-functions.sh

```diff
@@ -12,6 +12,29 @@ readonly KEEPALIVED_CONFIG="/etc/keepalived/keepalived.conf"
 readonly KEEPALIVED_DEFAULTS="/etc/sysconfig/keepalived"
 
 
+function cleanup() {
+  echo "  - Cleaning up ... "
+  [ -n "$1" ] && kill -TERM $1
+
+  local interface=$(get_network_device "$NETWORK_INTERFACE")
+  local vips=$(expand_ip_ranges "$HA_VIPS")
+  echo "  - Releasing VIPs ${vips} (interface ${interface}) ... "
+
+  local regex='^.*?/[0-9]+$'
+
+  for vip in ${vips}; do
```

Comment:
I think for cleanup we need to cleanup anything we created, but we *can*
fail if something really unexpected happens in our script.  So we should
try to clean up as much as we can - if get_network_device or
expand_ip_ranges fails, is there anything else we need to clean up?

On Wed, Jun 8, 2016 at 1:12 AM, Ram Ranganathan <notifications@github.com>
wrote:

> In images/ipfailover/keepalived/lib/failover-functions.sh
> <https://github.com/openshift/origin/pull/9214#discussion_r66193179>:
>
> > @@ -12,6 +12,29 @@ readonly KEEPALIVED_CONFIG="/etc/keepalived/keepalived.conf"
> >  readonly KEEPALIVED_DEFAULTS="/etc/sysconfig/keepalived"
> >
> >
> > +function cleanup() {
> > +  echo "  - Cleaning up ... "
> > +  [ -n "$1" ] && kill -TERM $1
> > +
> > +  local interface=$(get_network_device "$NETWORK_INTERFACE")
> > +  local vips=$(expand_ip_ranges "$HA_VIPS")
> > +  echo "  - Releasing VIPs ${vips} (interface ${interface}) ... "
> > +
> > +  local regex='^.*?/[0-9]+$'
> > +
> > +  for vip in ${vips}; do
>
> sure that would work too in lieu of the || : - just felt it cleaner to
> put it in there. As we don't care about errors (ala if the VIP was not
> assigned/already removed) - maybe a comment here would help make that
> clearer?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9214/files/8c268838f22a760d9e1b6730420f475cc4f8f564#r66193179>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p-GLcVPbUYVEGnjxnIXaERpH2Fv8ks5qJk85gaJpZM4IwllB>
> .
>